lib/sensors/include/URM04Sensor.h

```diff
@@ -21,41 +18,25 @@ class URM04Sensor {
   uint8_t startAddr;
   // command buffer
   uint8_t cmdst[10];
-  // state machine reading step
-  uint8_t readingStep;
-  // arrays needed to read and write distance from sensor
-  int urmID[urmAccount];
-  uint32_t urmData[urmAccount];
-  // timer related members
-  uint64_t managerTimer;
-  Timer clock;
   // UART protocol pins
   PinName RX;
   PinName TX;
-  // successful read
-  bool read_success;
-
-  // trigger the mesausrements from URM04
-  void urmTrigger(int id);
-  // reads the distance from URM04
-  void urmReader(int id);
-  // transmit commands
-  void transmitCommands();
-
-  // analyzes the distance**
-  void analyzeUrmData(uint8_t cmd[]);
-  // runs the sensor - starts giving commands to the sensor
-  void runUrm04();
-  // decodes URM04 data
-  void decodeURM04();
+  // serial
+  BufferedSerial serial;
+  // variable to check if success_read
+  bool success;
+  // sensor distance
+  float m_distance;
```

Comment:
success def shouldnt be a member variable. i understand that the code i sent u took that approach, but it's not good imo
also why is m_distance a member var. do multiple functions use it or something?